pkg/controller/logstash/services_test.go

```diff
@@ -0,0 +1,250 @@
+					},
+				},
+			}},
+		},
+		{
+			name: "Default service plus one",
+			logstash: logstashv1alpha1.Logstash{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "logstash",
+					Namespace: "test",
+				},
+				Spec: logstashv1alpha1.LogstashSpec{
+					Services: []logstashv1alpha1.LogstashService{{
+						Name: "test",
+						Service: commonv1.ServiceTemplate{
+							Spec: corev1.ServiceSpec{
+								Ports: []corev1.ServicePort{
+									{Protocol: "TCP", Port: 9500},
+								},
+							},
+						},
+					}},
+				},
+			},
+			wantSvc: []corev1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "logstash-ls-test",
+						Namespace: "test",
+						Labels: map[string]string{
+							"common.k8s.elastic.co/type":   "logstash",
+							"logstash.k8s.elastic.co/name": "logstash",
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion:         "logstash.k8s.elastic.co/v1alpha1",
+								Kind:               "Logstash",
+								Name:               "logstash",
+								Controller:         &trueVal,
+								BlockOwnerDeletion: &trueVal,
+							},
+						},
+					},
+					Spec: corev1.ServiceSpec{
+						Selector: map[string]string{
+							"common.k8s.elastic.co/type":   "logstash",
+							"logstash.k8s.elastic.co/name": "logstash",
+						},
+						Ports: []corev1.ServicePort{
+							{Protocol: "TCP", Port: 9500},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "logstash-ls-api",
+						Namespace: "test",
+						Labels: map[string]string{
+							"common.k8s.elastic.co/type":   "logstash",
+							"logstash.k8s.elastic.co/name": "logstash",
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion:         "logstash.k8s.elastic.co/v1alpha1",
+								Kind:               "Logstash",
+								Name:               "logstash",
+								Controller:         &trueVal,
+								BlockOwnerDeletion: &trueVal,
+							},
+						},
+					},
+					Spec: corev1.ServiceSpec{
+						Selector: map[string]string{
+							"common.k8s.elastic.co/type":   "logstash",
+							"logstash.k8s.elastic.co/name": "logstash",
+						},
+						ClusterIP: "None",
+						Ports: []corev1.ServicePort{
+							{Name: "api", Protocol: "TCP", Port: 9600},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			defaultSvc := defaultServiceFromOwner(&tc.logstash)
+			client := k8s.NewFakeClient(&tc.logstash, defaultSvc)
+			params := Params{
+				Context:  context.Background(),
+				Client:   client,
+				Logstash: tc.logstash,
+			}
+			haveSvc, err := reconcileServices(params)
+			require.NoError(t, err)
+			require.Equal(t, len(tc.wantSvc), len(haveSvc))
+
+			for i := range tc.wantSvc {
+				comparison.AssertEqual(t, &tc.wantSvc[i], &haveSvc[i])
+			}
+		})
+	}
+}
+
+func defaultServiceFromOwner(owner *logstashv1alpha1.Logstash) *corev1.Service {
```

Comment:
nit: why do we need a function to create the default service while it should be created automatically?

Reply:
Good point - I've removed it